pkg/console/subresource/consoleserver/types.go

```diff
@@ -111,10 +111,40 @@ type ProjectAccess struct {
 	AvailableClusterRoles []string `yaml:"availableClusterRoles,omitempty"`
 }
 
+// CatalogTypesState defines the state of the catalog types based on which the types will be enabled or disabled.
+type CatalogTypesState string
+
+const (
+	CatalogTypeEnabled  CatalogTypesState = "Enabled"
+	CatalogTypeDisabled CatalogTypesState = "Disabled"
+)
+
+// DeveloperConsoleCatalogTypesState defines the state of the sub-catalog types.
+type DeveloperConsoleCatalogTypesState struct {
```

Comment:
Hmm, missed that in the API change, but this defines the "Catalog Types", and contains a state, so I believe we should drop the state suffix from this type definition, or?

```suggestion
// DeveloperConsoleCatalogTypes defines the state of the sub-catalog types.
type DeveloperConsoleCatalogTypes struct {
```

Reply:
here I am trying to define the state of the catalog types

Reply:
I was thinking that `State CatalogTypesState `yaml:"state,omitempty"`` is "the state" and this is "more", but its fine, we can keep this.